app/models/shipit/stack.rb

```diff
@@ -386,16 +389,21 @@ def active_task
     end
 
     def locked?
+      lock_reason.present? && lock_level != 'advisory'
+    end
+
+    def soft_locked?
       lock_reason.present?
     end
 
-    def lock(reason, user)
-      params = {lock_reason: reason, lock_author: user}
+    def lock(reason, user, lock_level: 'enforced')
+      lock_level ||= 'enforced'
```

Comment:
What's the best way to set a default value for `lock_level` ? I used a keyword argument since it's a more optional argument, but could've used `def lock(reason, user, lock_level = 'enforced')` also. I also had to add the variable assignation above since the argument could either be present or sent as `nil`, and we don't want a `nil` value. But this looks redundant.

What do you think ?

Reply:
Same here I'd reference a constant.

As for the keyword arg, yeah it's fine like this.